extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {ContainWrapper} from '../../../src/preact/component';
+import {forwardRef} from '../../../src/preact/compat';
+import {setStyle} from '../../../src/style';
+import {
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from '../../../src/preact';
+import {useStyles} from './sidebar.jss';
+
+const ANIMATION_DURATION = 350;
+const ANIMATION_PRESETS = {
+  fadeIn: [{opacity: 0}, {opacity: 1}],
+  slideInLeft: [{transform: 'translateX(-100%)'}, {transform: 'translateX(0)'}],
+  slideInRight: [{transform: 'translateX(100%)'}, {transform: 'translateX(0)'}],
+};
+
+const ANIMATION_EASE_IN = 'cubic-bezier(0,0,.21,1)';
+
+/**
+ * @param {T} current
+ * @return {{current: T}}
+ * @template T
+ */
+function useValueRef(current) {
+  const valueRef = useRef(null);
+  valueRef.current = current;
+  return valueRef;
+}
+
+/**
+ * @param {!AccordionDef.Props} props //TODO: will update
+ * @param ref
+ * @return {PreactDef.Renderable}
+ */
+function SidebarWithRef(
+  {
+    as: Comp = 'div',
+    side = 'left',
```

Comment:
Possible ok for the first PR, but does the previous versions of sidebar refer to this as left|right? Or something more LTR/RTL neutral?

Reply:
Good point.  Yes the previous version of sidebar refers to this as "left" and "right".